src/player.cpp

```diff
@@ -10776,6 +10782,54 @@ activate your weapon."), gun->tname().c_str(), _(mod->location.c_str()));
     }
 }
 
+bool player::invoke_item( item* used )
+{
+    if( !has_enough_charges( *used, true ) ) {
+        return false;
+    }
+        
+    if( used->type->use_methods.size() < 2 ) {
+        return used->type->invoke( this, used, pos() );
+    }
+
+    uimenu umenu;
+    umenu.text = string_format( _("What to do with your %s?"), used->tname().c_str() );
+    int num_total = 0;
+    for( const auto &um : used->type->use_methods ) {
+        if( um.get_actor_ptr() == nullptr ) {
+            debugmsg( "Multi-use items with cpp iuse_functions not supported." );
+            return false;
+        }
+
+        bool usable = um.get_actor_ptr()->can_use( this, used, false, pos() );
+        const std::string &aname = item_action_generator::generator().get_action_name( um );
+        umenu.addentry( num_total, usable, MENU_AUTOASSIGN, aname.c_str() );
```

Comment:
There is an overloaded version of `addentry` that takes a proper `std::string`, so you don't need to supply a `const char*` - call it without the `.c_str()`.

Giving a `const char *` invokes the string formating, if `aname` contains a format specifier like "%s", the function will try to access the parameters behind the format and there are none, that's bad.